modules/sfp_haveibeenpwned.py

```diff
@@ -135,23 +179,54 @@ def handleEvent(self, event):
             self.results[eventData] = True
 
         data = self.query(eventData)
-        if data is None:
+        if data is not None:
+            for n in data:
+                try:
+                    if not self.opts['api_key']:
+                        site = n["Title"]
+                    else:
+                        site = n["Name"]
+                except Exception as e:
+                    self.sf.debug(f"Unable to parse result from HaveIBeenPwned?: {e}")
+                    continue
+
+                evt = eventName + "_COMPROMISED"
+                # Notify other modules of what you've found
+                e = SpiderFootEvent(evt, eventData + " [" + site + "]",
+                                    self.__name__, event)
+                self.notifyListeners(e)
+
+        pasteData = self.queryPaste(eventData)
+        if pasteData is None:
             return None
 
-        for n in data:
+        sites = {
+            "Pastebin": "https://pastebin.com/",
+            "Pastie": "http://pastie.org/",
+            "Slexy": "https://slexy.org/",
+            "Ghostbin": "https://ghostbin.com/",
+            "QuickLeak": "http://www.quickleak.net/",
+            "JustPaste": "https://justpaste.it/",
+        }
+
+        for n in pasteData:
             try:
-                if not self.opts['api_key']:
-                    site = n["Title"]
-                else:
-                    site = n["Name"]
+                source = n.get("Source")
+                site = source
+                if source in sites.keys():
+                    site = sites[n.get("Source")]
+
+                if not n.get("Source") is None:
+                    e = SpiderFootEvent("LEAKSITE_URL", site, self.__name__, event)
+                    self.notifyListeners(e)
+
+                if not n.get("Title") is None:
+                    e = SpiderFootEvent("LEAKSITE_CONTENT", n.get("Title"), self.__name__, event)
```

Comment:
Look at sfp_pastebin to see how `LEAKSITE_CONTENT` should be reported.

Reply:
The whole of res['content'] is being reported, in this case should it be str(pasteData) ?